### DISCUSSION TITLE
Dynamic import working

### DISCUSSION THREAD

**Discussion by K-Kumar-01** [vercel/next.js#20692] [Help]
I have used dynamic import in one of my [projects.](https://github.com/k-kumar-01/frontend)
Wanted to know a bit about the component.
When we dynamically import a component say 
```
const DynamicComp = dynamic(()=>import("./dynamicComp"));
```
and then render it directly like
```
const MyComp = () => <React.Fragment>
    // some components up here
    <DynamicComp/>
</React.Fragment>
```

The `DyanmicComp` is not in DOM viewport so will it be imported when `MyComp` is mounted/rendered or will it be imported when the viewport is close to the `DynamicComp`.

**Comment by michrome** [comment]
When MyComp is mounted/rendered - it's not like a lazy-loaded image.

**Reply by K-Kumar-01** [reply]
So suppose if I want to load the component when it is near DOM viewport how can I do that.
I have heard that it increases performance as the code becomes optimized.
I know that `React.lazy` works but it gave an error in Nextjs.
Any solution?

**Comment by K-Kumar-01** [comment]
@michrome 
Is there a way to dynamic import functions that are not React components in nextjs?
Basically, I want to import a function that will call an API only if a certain button is clicked.
Is there any way to do this?

**Reply by michrome** [reply]
You could use a [JavaScript dynamic import](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/import#Dynamic_Imports) but it's worth reading the caveats: static imports are preferable for loading initial dependencies, and can benefit more readily from static analysis tools and tree shaking.
